pkg/prometheus/promcfg.go

```diff
@@ -224,12 +224,28 @@ func (cg *configGenerator) generateConfig(
 	var scrapeConfigs []yaml.MapSlice
 	for _, identifier := range sMonIdentifiers {
 		for i, ep := range sMons[identifier].Spec.Endpoints {
-			scrapeConfigs = append(scrapeConfigs, cg.generateServiceMonitorConfig(version, sMons[identifier], ep, i, apiserverConfig, basicAuthSecrets, bearerTokens, p.Spec.OverrideHonorLabels))
+			scrapeConfigs = append(scrapeConfigs,
+				cg.generateServiceMonitorConfig(
+					version,
+					sMons[identifier],
+					ep, i,
+					apiserverConfig,
+					basicAuthSecrets,
+					bearerTokens,
+					p.Spec.OverrideHonorLabels,
+					p.Spec.EnforcedNamespaceLabel))
```

Comment:
not part of this PR, but I think we have already many precedences wher we could benefit from the "functional API" idiom instead of having very long constructor methods?

Reply:
Agreed!